app/javascript/controllers/autoresize_controller.ts

```diff
@@ -3,16 +3,31 @@ import { attach } from '@frsource/autoresize-textarea';
 import { isTextAreaElement } from '@coldwired/utils';
 
 export class AutoresizeController extends ApplicationController {
+  declare observer: IntersectionObserver;
+
   #detach?: () => void;
   connect(): void {
     if (isTextAreaElement(this.element)) {
-      this.#detach = attach(this.element)?.detach;
       this.element.classList.add('resize-none');
+      this.observer = new IntersectionObserver(this.onIntersect.bind(this), {
+        threshold: [0]
+      });
+      this.observer.observe(this.element);
+    }
+  }
+
+  onIntersect(entries: IntersectionObserverEntry[]): void {
+    const visible: boolean = entries[0].isIntersecting == true;
```

Comment:
```suggestion
    const visible = entries[0].isIntersecting == true;
```
(ts dérive bien le type non ?)